src/polygon.cpp

```diff
@@ -502,32 +502,26 @@ class EarClip {
         return totalCost;
       }
 
-      Box earBox = Box{vec3(center.x - radius, center.y - radius, 0),
-                       vec3(center.x + radius, center.y + radius, 0)};
-      earBox.Union(vec3(pos, 0));
-      collider.collider.Collisions(VecView<const Box>(&earBox, 1),
-                                   collider.ind);
+      Rect earBox = Rect(vec2(center.x - radius, center.y - radius),
+                         vec2(center.x + radius, center.y + radius));
+      earBox.Union(pos);
+      earBox.min -= epsilon;
+      earBox.max += epsilon;
 
       const int lid = left->mesh_idx;
       const int rid = right->mesh_idx;
-
-      totalCost = transform_reduce(
-          countAt(0), countAt(collider.ind.size()), totalCost,
-          [](double a, double b) { return std::max(a, b); },
-          [&](size_t i) {
-            const VertItr test = collider.itr[collider.ind.Get(i, true)];
-            if (!Clipped(test) && test->mesh_idx != mesh_idx &&
-                test->mesh_idx != lid &&
-                test->mesh_idx != rid) {  // Skip duplicated verts
-              double cost = Cost(test, openSide, epsilon);
-              if (cost < -epsilon) {
-                cost = DelaunayCost(test->pos - center, scale, epsilon);
-              }
-              return cost;
-            }
-            return std::numeric_limits<double>::lowest();
```

Comment:
Is this case handled inside the query function now?

Reply:
I think for lowest, it just means it will not change the cost (because we take the max).